File: 868. Binary Gap/solution.py
class Solution(object):
  def binaryGap(self, n):
    """
    :type n: int
    :rtype: int
    """
    lastOne = 32
    result = 0
    for index in range(32):
      if n & (1 << index):
        if index - lastOne > result:
          result = index - lastOne
        lastOne = index
    return result

File: 868. Binary Gap/solution.java
class Solution {
  public int binaryGap(int n) {
    int lastOne = 32;
    int result = 0;
    for (int index = 0; index < 32; index++){
      if ((n & (1 << index)) != 0){
        if (index - lastOne > result){
          result = index - lastOne;
        }
        lastOne = index;
      }
    }
    return result;
  }
}